src/librustc/middle/dependency_format.rs

```diff
@@ -103,6 +103,10 @@ pub fn calculate(sess: &session::Session) {
 
 fn calculate_type(sess: &session::Session,
                   ty: config::CrateType) -> DependencyList {
+    if sess.opts.output_types.contains_key(&config::OutputType::Metadata) {
```

Comment:
Here I think we could just say that if the outputs contain `link` we can take the paths below, right?